### PR TITLE
Remove old Roslyn suppression code

### LINKED ISSUE #2927: Remove old Roslyn suppression code
### Description

The existing Roslyn suppression handling code can be removed, along with the code that handles the SonarAnalyzer NuGet packages being installed in the MSBuild project (see #2925).

TODO
- [ ] delete the existing Roslyn suppression handing code
- [ ] check CFamily suppressions are not affected
- [ ] if possible, remove the reference to the SonarAnalyzer assembly from the SLVS code and MSBuild projects

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #2927

### CODE REVIEW

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt [hunk 1]
```diff
@@ -94,8 +94,6 @@ Assembly: 'SonarLint.2022, Version=6.0.0.0, Culture=neutral, PublicKeyToken=c5b6
 Relative path: 'SonarLint.2022.dll'
 
 Referenced assemblies:
-- 'Microsoft.CodeAnalysis, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:97]: We no longer have code that directly references the SonarAnalyzer callbacks, which means we don't need to reference the MS CodeAnalysis assemblies either.

File: src/Integration.Vsix/Integration.Vsix.csproj [hunk 2]
```diff
@@ -125,11 +122,6 @@
     <Reference Include="Microsoft.CSharp" />
   </ItemGroup>
 
-  <PropertyGroup>
-    <!-- Suppress NuGet warning about the explicit downgrade to System.Collections.Immutable v1.1.37 -->
-    <NoWarn>NU1605</NoWarn>
-  </PropertyGroup>
-
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/Integration.Vsix.csproj:132]: We don't reference the MS.CodeAnalysis NuGet packages, which means we don't need to reference specific versions of the `System.Collections.Immutable` or `System.Reflection.Metadata` assemblies either.

That in turn means we no longer need to suppress this NuGet warning.

**Comment by duncanp-sonar** [comment]
There are two commits:
* the first just deletes the old code.
* the second updates the project file, package lock files and asmrefs.

I've checked that the contents of the VSIX haven't changed - the set of files we ship is still the same, even though we no longer reference the SonarAnalyzer assembly directly.

I also manually tested that C++ suppressions still work (or at least, work as well as they do currently).

**Review by rita-sonarsource — APPROVED** [review]
Woohoo!

**Comment by duncanp-sonar** [comment]
FYI 

**Before (LOC, #Types)**
image

**After (LOC, #Types)**
image


= -561 lines of product code (6.0% reduction)
= -1492 lines of test code (9.3% reduction)

**Pull request merged**